save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/TestExecutionController.kt

```diff
@@ -100,6 +101,25 @@ class TestExecutionController(private val testExecutionService: TestExecutionSer
                 }
             }
 
+    /**
+     * @param executionId an ID of Execution to group TestExecutions
+     * @param authentication
+     * @return a list of [TestExecutionDto]s
+     */
+    @GetMapping(path = ["/api/$v1/testLatestExecutionsPieChart"])
+    @Suppress("TYPE_ALIAS")
+    fun getDataPieChartTestExecutions(
+        @RequestParam executionId: Long,
+        authentication: Authentication,
+    ): Mono<List<DataPieChart>> =
+            justOrNotFound(executionService.findExecution(executionId)).filterWhen {
+                projectPermissionEvaluator.checkPermissions(authentication, it, Permission.READ)
+            }.mapNotNull {
+                testExecutionService.getTestExecutions(executionId).groupBy { it.test.testSuite.name }.map { (testSuiteName, testExecutions) ->
+                    DataPieChart(testSuiteName, testExecutions.count())
+                }
+            }
```

Comment:
```suggestion
            }.flatMapIterable {
                testExecutionService.getTestExecutions(executionId).groupBy { it.test.testSuite.name }
            }.map { (testSuiteName, testExecutions) ->
                DataPieChart(testSuiteName, testExecutions.count())
            }
```
This way you won't need `Mono<List<>>` and will have less nested lambdas